### PR TITLE
feat(scheduler-targets-alpha): `KinesisStreamPutRecord` Target

### LINKED ISSUE #27451: (aws-scheduler-targets-alpha): Add KinesisStreamPutRecord Target
### Describe the feature

Work to support L2 constructs for AWS Scheduler is in progress (https://github.com/aws/aws-cdk/issues/23394). See the approved [RFC](https://github.com/aws/aws-cdk-rfcs/pull/486/files). RFC planned to add 12 templates targets, but only Lambda Invoke is currently implemented (https://github.com/aws/aws-cdk/pull/26575). 

This issue tracks implementation of  `KinesisStreamPutRecord` target to put a record to a Kinesis Stream.

### Use Case

Customers would like to use templated target `KinesisStreamPutRecord ` to be able to put a record to a Kinesis Stream on schedule. L2 target construct should grant required permissions to the `AWS Scheduler` to put records to a Kinesis Stream.  

### Proposed Solution

The proposed solution needs to be adopted to the recent examples of `LambdaInvoke` (https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-scheduler-targets-alpha/lib/lambda-invoke.ts). 

Solution should also include unit and integration tests. 

Class `KinesisStreamPutRecord` should: 
1. Grant Scheduler Execution Role permissions to Put Records via `addTargetActionToRole`
2. Override `bindBaseTargetConfig` to return `kinesisParameters`  as part of `ScheduleTargetConfig`.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.99.1

### Environment details (OS name and version, etc.)

MacOS

### PR DESCRIPTION
**Pull request by go-to-k**
This PR adds KinesisStreamPutRecord Target for EventBridge Scheduler.

Closes #27451.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-scheduler-targets-alpha/lib/kinesis-stream-put-record.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { sameEnvDimension } from './util';
+
+/**
+ * Properties for a Kinesis Data Streams Target
+ */
+export interface KinesisStreamPutRecordProps extends ScheduleTargetBaseProps {
+  /**
+   * The shard to which EventBridge Scheduler sends the event.
+   *
+   * A length of `partitionKey` must be between 1 and 256.
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-scheduler-targets-alpha/lib/kinesis-stream-put-record.ts:15]:
```suggestion
   * The length must be between 1 and 256.
```

File: packages/@aws-cdk/aws-scheduler-targets-alpha/test/kinesis-stream-put-record.test.ts [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+    const streamTarget = new KinesisStreamPutRecord(importedStream, {
+      partitionKey: 'key',
+    });
+
+    expect(() =>
+      new Schedule(stack, 'MyScheduleDummy', {
+        schedule: expr,
+        target: streamTarget,
+      })).toThrow(/Both the schedule and the stream must be in the same region/);
+  });
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-scheduler-targets-alpha/test/kinesis-stream-put-record.test.ts:333]:
```suggestion
  test.each([
    ['account', 'arn:aws: kinesis:us-east-1:999999999999:stream/Foo', /Both the schedule and the stream must be in the same account./],
    ['region', 'arn:aws: kinesis:eu-central-1:123456789012:stream/Foo', /Both the schedule and the stream must be in the same region./],
  ])('throws when Kinesis Data Stream is imported from different %s', (_, arn: string, expectedError: RegExp) => {
    const importedStream = kinesis.Stream.fromStreamArn(stack, 'ImportedStream', arn);
    const streamTarget = new KinesisStreamPutRecord(importedStream, {
      partitionKey: 'key',
    });
    expect(() =>
      new Schedule(stack, 'MyScheduleDummy', {
        schedule: expr,
        target: target,
      })).toThrow(expectedError);
  });
```
`fromStreamArn` should use the same stack.
[`fromStreamAttributes`](https://github.com/aws/aws-cdk/blob/02957dfddd895ffcca1dd485da312544babf6ccb/packages/aws-cdk-lib/aws-kinesis/lib/stream.ts#L738) will require to solve the environment for the imported stream ARN:
```
  public static fromStreamAttributes(scope: Construct, id: string, attrs: StreamAttributes): IStream {
    class Import extends StreamBase {
      public readonly streamArn = attrs.streamArn;
      public readonly streamName = Stack.of(scope).splitArn(attrs.streamArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
      public readonly encryptionKey = attrs.encryptionKey;
    }

    return new Import(scope, id, {
      environmentFromArn: attrs.streamArn,
    });
  }
```
Also, can you please add unit tests to verify that the environment gets resolved correctly when using `fromStreamArn` and `fromStreamAttributes` after this change?

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Looks good overall, thanks 👍
The unit tests on imported streams will require a fix on the importing function.

**Comment by go-to-k** [comment]
@lpizzinidev 

I just changed! Please check them.

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Review by vinayak-kukreja — DISMISSED** [review]
Hey @go-to-k , thank you for this contribution.

**Comment by go-to-k** [comment]
@vinayak-kukreja 

Thanks for your approval! I fixed conflicts. Please approve again.

I fixed [the other PR](https://github.com/aws/aws-cdk/pull/27842) the same way, but when one of them gets merged, the other one gets conflicts again. If that happens, I will fix it again.

**Pull request merged**